src/Compilers/CSharp/Portable/Binder/LocalBinderFactory.cs

```diff
@@ -946,7 +946,17 @@ public override void VisitLocalDeclarationStatement(LocalDeclarationStatementSyn
         public override void VisitVariableDeclarator(VariableDeclaratorSyntax node)
         {
             Visit(node.ArgumentList);
-            Visit(node.Initializer?.Value);
+
+            if (node.Initializer is { } initializer)
+            {
+                var oldEnclosing = _enclosing;
+                var localInProgressBinder = new LocalInProgressBinder(initializer, _enclosing);
+                AddToMap(initializer, localInProgressBinder);
+
+                _enclosing = localInProgressBinder;
+                Visit(initializer.Value);
```

Comment:
>Visit [](http://example.com/codeflow?start=16&length=5)

We probably have an overload that takes an enclosing binder as parameter and deals with saving/setting/restoring the context.